src/components/AmountEditable.tsx

```diff
@@ -320,12 +323,44 @@ export const AmountEditable: ParentComponent<{
                 <div class={DIALOG_POSITIONER}>
                     <Dialog.Content
                         class={DIALOG_CONTENT}
-                        onEscapeKeyDown={() => setIsOpen(false)}
+                        onEscapeKeyDown={() => {
```

Comment:
looks like this logic is identical to the close button, could probably pull it out as a function that gets used in both?